### PR TITLE
[TASK] Move the unit tests from Travis CI to GitHub actions

### PR DESCRIPTION
**Pull request by oliverklee**
Removing the Travis CI integration will follow in a separate commit.

Part of #818

### CODE REVIEW

**Comment by oliverklee** [comment]
Currently, the Travis build status is not automatically visible in this PR.
Here's the direct link to the build:
https://travis-ci.org/github/MyIntervals/emogrifier/builds/667714204

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -97,3 +97,51 @@ jobs:
...
+      - name: Cache dependencies installed with composer
+        uses: actions/cache@v1
+        with:
+          path: ~/.composer/cache
+          key: php${{ matrix.php-version }}-${{ matrix.dependencies }}-composer-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            php${{ matrix.php-version }}-${{ matrix.dependencies }}-composer-
+
+      - name: Install Composer dependencies
+        run: |
```
Comment (JakeQZ) [inline .github/workflows/ci.yml]: Might a `composer install` step be needed first, before `composer update`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -97,3 +97,51 @@ jobs:
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v1
+        with:
+          php-version: ${{ matrix.php-version }}
+          ini-values: xdebug.max_nesting_level=500
```
Comment (JakeQZ) [inline .github/workflows/ci.yml]: Do we need this (we did not previously change the setting)?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -97,3 +97,51 @@ jobs:
...
+          path: ~/.composer/cache
+          key: php${{ matrix.php-version }}-${{ matrix.dependencies }}-composer-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            php${{ matrix.php-version }}-${{ matrix.dependencies }}-composer-
+
+      - name: Install Composer dependencies
+        run: |
+          if [[ "${{ matrix.dependencies }}" == 'lowest' ]]; then
+            DEPENDENCIES='--prefer-lowest';
+          fi;
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply .github/workflows/ci.yml]: Yes, that would be cleaner. I'll update the PR accordingly.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -97,3 +97,51 @@ jobs:
...
+          composer update --no-progress "${DEPENDENCIES}";
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply .github/workflows/ci.yml]: Exactly. (This kind of problem will go away once we've moved the tools from Composer dependencies to Phive.)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (oliverklee) [reply .github/workflows/ci.yml]: No, this is not required (and would slow down the process as we'd then potentially download two versions of some packages). `update` always includes the `install` (and for calculating the updates, Composer does not need any packages installed.)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (oliverklee) [reply .github/workflows/ci.yml]: Yes. Without this setting, we'd get a "maximum nesting level reached" error in `CssInlinerTest::inlineCssAppliesMoreSpecificCssSelectorToMatchingElements`:
https://github.com/MyIntervals/emogrifier/runs/539483776?check_suite_focus=true

File: .github/workflows/ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (oliverklee) [reply .github/workflows/ci.yml]: Done.

**Pull request merged**